### PR TITLE
Maro geo vis

### PR DESCRIPTION
**Pull request by Meroy9819**
# Description

Env-geographic tool with real-time mode and local mode. Contains front-end package(not raw code), back-end code and deploy scripts in this PR.
## Linked issue(s)/Pull request(s)

- [issue_number](issue_link)

## Type of Change

- [ ] Non-breaking bug fix
- [ ] Breaking bug fix
- [x] New feature
- [ ] Test
- [ ] Doc update
- [ ] Docker update

## Related Component

- [x] Simulation toolkit
- [ ] RL toolkit
- [ ] Distributed toolkit

## Has Been Tested

- OS:
  - [x] Windows
  - [ ] Mac OS
  - [ ] Linux
- Python version:
  - [x] 3.6
  - [ ] 3.7
- Key information snapshot(s):

## Needs Follow Up Actions

- [ ] New release package
- [ ] New docker image

## Checklist

- [ ] Add/update the related comments
- [ ] Add/update the related tests
- [ ] Add/update the related documentations
- [ ] Update the dependent downstream modules usage

### CODE REVIEW

File: scripts/launch_realtime_vis.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+1. launch the servcies' docker-compose.yml if services not started.
+
+2. lauch the experiment file
+
+
+"""
+
+
+import docker
```
Comment (zhawan) [inline scripts/launch_realtime_vis.py:27]: only 1 line, seems strange

File: docs/source/key_components/geographic_visualization.rst [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+To send data to database, user need to set the value of the environment variable
+"MARO_STREAMIT_ENABLED" to "true". If user wants to specify the experiment name,
+set the environment variable "MARO_STREAMIT_EXPERIMENT_NAME". If user does not 
+set this value, a unique experiment name would be processed automatically. User
+could check the experiment name through database. It should be noted that when
+selecting a topology, user must select a topology with specific geographic
+information. The experimental data obtained by using topology files without
+geographic information cannot be used in the Env-geographic tool.
+
+.. code-block:: sh
```
Comment (zhawan) [inline docs/source/key_components/geographic_visualization.rst:58]: seems shall be python

File: docs/source/key_components/geographic_visualization.rst [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+Currently, Env-geographic supports two modes, which are real-time mode and local mode.
+The former mode shows users real-time experimental data and helps users judge the effectiveness of the model.
+The latter mode allows users to freely view the data after the experiment,
+helping users to view the effect of the experiment and make subsequent decisions.
+
+
+Dependency
+----------
+
+Env-geographic's startup depends on docker.
```
Comment (Jinyu-W) [inline docs/source/key_components/geographic_visualization.rst:15]: attach the link to docker installation here?

File: docs/source/key_components/geographic_visualization.rst [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+How to Use?
+-----------
+
+Env-geographic has 3 parts: front-end, back-end and experiment database. To start this tool,
+user need to start the docker containers, then start an experiment. The experimental data would
+send to database automatically.
+
+Start service
+~~~~~~~~~~~~~
+In order to start the env-geographic service, user need to start 4 docker
```
Comment (Jinyu-W) [inline docs/source/key_components/geographic_visualization.rst:28]: I find 4 docker files in this PR. should we build the docker images in advance?

File: docs/source/key_components/geographic_visualization.rst [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+maro_vis_front_end, questdb/questdbwith the following command:
+
+.. code-block:: sh
+
+    maro inspector geo
+
+----
+
+After the command is executed successfully, user
+could view the front_end page through localhost:8080
```
Comment (Jinyu-W) [inline docs/source/key_components/geographic_visualization.rst:39]: Note that the default ports are ... and mention how to change the port below

File: docs/source/key_components/geographic_visualization.rst [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+
+Env-geographic's startup depends on docker. 
+Therefore, users need to install docker on the machine and ensure that it can run normally.
+
+
+How to Use?
+-----------
+
+Env-geographic has 3 parts: front-end, back-end and experiment database. To start this tool,
+user need to start the docker containers, then start an experiment. The experimental data would
```
Comment (Jinyu-W) [inline docs/source/key_components/geographic_visualization.rst:23]: start containers manually or through maro cli?

File: docs/source/key_components/geographic_visualization.rst [hunk 7]
```diff
@@ -0,0 +1,156 @@
...
+
+
+Dependency
+----------
+
+Env-geographic's startup depends on docker. 
+Therefore, users need to install docker on the machine and ensure that it can run normally.
+
+
+How to Use?
```
Comment (Jinyu-W) [inline docs/source/key_components/geographic_visualization.rst:19]: Actually, I still don't understand how to use it Orz

File: docs/source/key_components/geographic_visualization.rst [hunk 8]
```diff
@@ -0,0 +1,201 @@
...
+
+Env-geographic's startup depends on docker. 
+Therefore, users need to install docker on the machine and ensure that it can run normally.
+User could get docker through `Docker installation <https://docs.docker.com/get-docker/>`_.
+
+
+How to Use?
+-----------
+
+Env-geographic has 3 parts: front-end, back-end and experiment database. To start this tool,
```
Comment (Jinyu-W) [inline docs/source/key_components/geographic_visualization.rst:23]: Users need 2 steps to start this tool:
1. start the database and choose the experiment name;
2. start the service with `maro inspector geo` in command line.

Then, organize the content like below?

Start database
\~~~~~~~~~~

Choose an existing experiment
^^^^^^^^^^^^^^^^^^^^^^

  Create a new experiment
^^^^^^^^^^^^^^^^^^

Start service
\~~~~~~~~~~

**Review by Jinyu-W — APPROVED** [review]

**Pull request merged**